src/peripherals/dma.ts

```diff
@@ -549,4 +587,13 @@ export class RPDMA extends BasePeripheral implements Peripheral {
     this.rp2040.setInterrupt(IRQ.DMA_IRQ0, !!this.intStatus0);
     this.rp2040.setInterrupt(IRQ.DMA_IRQ1, !!this.intStatus1);
   }
+
+  sniffTransfer(channel: RPDMAChannel, datum: number) {
+    const sniffChannel = (this.sniffCtrl >> DMACH_SHIFT) & DMACH_MASK;
+    if ((this.sniffCtrl & EN) && (channel.index === sniffChannel)) {
+      // TODO: Actually obey settings in sniffCtrl...
+      // currently assuming OUT_INV = OUT_REV = SNIFF_BSWAP = 0; CALC = 0xf
+      this.sniffData += datum;
```

Comment:
Probably better to cut at 32 bits (in JavaScript, all numbers are stored as double precision floating point by default):

```js
this.sniffData = (this.sniffData + datum) >>> 0;
```

Reply:
Oops, good catch. I didn't think about this at the moment – for the ADC/DMA register fiddling from Micropython, `uctypes` did this for me ;)